README
Project Adviser:
Walter Bender - walter@laptop.org

Core Developer(s):	
Arjun Sarwal - arjun@laptop.org

Contributor(s):
Many many!
See http://wiki.laptop.org/go/Measure

Contact:
Email: arjun@laptop.org
IRC  : arjs on Freenode




NEWS
103

* Update screenshots (sdziuda),
* Fix sensor toolbar interval icon (Martin Abente Lahaye),
* Fix access to rate value (Martin Abente Lahaye),
* Fix tuning toolbar icons (Martin Abente Lahaye),
* Fix tuning toolbar positioning and note buttons (Martin Abente Lahaye),
* Make tuning instrument notes audible (Martin Abente Lahaye),
* Port to Python 3 (Srevin Saju),
* Remove empty translations (James Cameron),
* New translations (mbouzada, Paulo Francisco, Besnik_b),
* Regenerate POT file (James Cameron),

102

* Fix for Gst.Pipeline arguments (Walter Bender),
* Modernise audiograb (James Cameron),
* Regenerate POT file (James Cameron),
* New translations (Markus, Martín Oesterreich, Paulo Francisco, Besnik Bleta),
* Fix flake8 errors (James Cameron),
* Update metadata and license details (James Cameron, Vipul Gupta),
* Port from GObject to GLib (James Cameron),
* Remove unused GConf reference (James Cameron),

101

* New translations (Chris Leonard et al),
* On Fedora 18 GStreamer 1.0.10 systems, display an error (James Cameron).

100

* Port to Gtk+ 3.0, GStreamer 1.0, for Ubuntu 16.04 and later (James Cameron)

53

* New translations
* Added repository to activity.info

52

ENHANCEMENT:
* New translations

BUG FIX:
* Restore mic mixer state on exit (James Cameron)


51

ENHANCEMENT:
* New translations

50

CODE CLEANUP:
* Remove duplicate hardware check (Alan Aguiar)
* Set default channels to 2 (Alan Aguiar)

49

ENHANCEMENT:
* Use csound for tone generation

BUG FIX:
* Add missing constant definition


48

ENHANCEMENT:
* New translations

47

ENHANCEMENTS:
* Ensure toolbars fit on narrow displays
* Use button palette, radio buttons instead of combo boxes

46

ENHANCEMENT:
* New translation

BUG FIX:
* Fix problem with sensor toolbars on XO4

45

ENHANCEMENTS:
* OLPC XO 4 support

44

BUG FIXES:
* Work around recent changes to numpy library that caused FFT to stop working
* Forcing XO 1.75, 3.0, and 4.0 to single channel
  (until such time as multi-channel capture works reliably)
* Fix problem with mic boost on 1.75 (#4288)

43

ENHANCEMENTS:
* New translations

42

ENHANCEMENTS:
* New translations
* Added cavaco (cavaquinho)
* Switch to freq. base if harmonics button pushed
* Blank trace while playing tone
* Define/save/restore custom instruments
* Match size of screen capture to size of trace window

BUG FIXES:
* Add support for old systems that lack subprocess.check_output method
* Fixed regression for mode switching when deboucing radio buttons

41

ENHANCEMENTS:
* Move time/frequency-base control to main toolbar
* Autoselect frequency base when tuing

40

ENHANCEMENTS:
* New icon for tuning toolbar by Manuel Quiñones
* Added Charango, Sitar, Ukulale, et al.
* New translations
* Add entry for user-specified tuning line
* Add entry for individual notes

39

ENAHNCEMENTS:
* Add tuning lines for guitar, violin, viola, cello, and bass (#3657)
* Channel mute (#3680)

BUG FIX:
* Typo preventing resistance mode from working on XO 1.75 hardware

38

ENHANCEMENT:
* Improved calibration for XO1.75 by Tony Forster

37

ENHANCEMENTS:
* New translations

BUG FIXES:
* Lock mode-selection radio buttons while mode is being reset (#3616).
* Fixed logic error that was causing triggering on both rising and
  falling edges (#2203).
* Fixed typo that had caused triggers to stop working (also #2203).
* Replaced deprecated "commands" with "subprocess".
* Pause grabbing (releasing audio device) when activity is not active.

36

ENHANCEMENTS:
* stereo capture
* calibrated data logging
* multi-channel data logging
* time-stamped data logging
* reconfigured toolbars to better reflect common use

BUG FIXES:
* screen capture works again
* journal spam eliminated

35

ENHANCEMENTS:
* new translations

BUG FIX:
* fixed problem with pootle that was preventing new strings to be committed

34 

* new translations
* new icons

33

* fix path problem for interval graphic on sound toolbar (#2730)
* removed dependency on HAL for determining hardware

32

* added missing tooltip to sensor interval spinner (#2121)
* updated depreciated activity.info fields (#2178)

31
* avoid race condition that leads to image capture corruption (#2119) (thanks to quozl)
* eliminate exception handler for tooltip text (#2121)
* increment capture file id in Journal (#2120)
* simplified Journal interactions (#2124)
* disabling sharing until sharing works again (#2123)

30
* eliminate dependency on amixer (thanks to Sascha Silbe)
* tweaked parameters for XO 1.5
* cleaned up code for non-OLPC hardware
* eliminate sensor toolbar for non-XO hardware
* improved hardware-detection logic
* added 1/10 second sample interval for sensor data
* added additional bias control for sensor data
* added tooltips for sliders (thanks to POSSE team)
* improved labels for capture intervals (thanks to POSSE team)
* added support for new toolbars
* moved frequency slider to main toolbar
* fixed problem with overcrowded toolbar preventing trigger control to appear
* new .es translations (thanks to rgs)

29
* support for XO 1.5 (thanks to Sayamindu Dasgupta)

28
* numerous patches to speed up code by Benjamin and Sebastian Berg

27
* added exception handling for gconf for old builds

26
* fixed toolbar display bug
* cleaned up sound record code
* caught exception causing resume from journal crash
* added display of sample value to toolbar

25
* Added invert button to invert time-based waveform
* Work around alsa problem on non-XO hardware that was preventing launch

24

23

21
* #8683 Fixed minor sensor logging problem

20
* Added license information

19
* New usage context based UI - sound and sensors context
* In sound context recording takes screenshots of waveforms and creates journal entries
* In sensors context recording writes csv values and creates journal entry 
* Textbox at bottom to show prompts for easy usage
* New even lower frequency display in sound context for 25hz waveforms
* Waveform color is XO stroke color

config.py
# Written by Arjun Sarwal <arjun@laptop.org>
# Copyright (C) 2007, Arjun Sarwal
# Copyright (C) 2009-11 Walter Bender
# Copyright (C) 2009, Benjamin Berg, Sebastian Berg
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# You should have received a copy of the GNU General Public License
# along with this library; if not, write to the Free Software
# Foundation, 51 Franklin Street, Suite 500 Boston, MA 02110-1335 USA


"""
Global configuration for Measure.
"""

import os
try:
    from sugar3.activity import activity
    MEASURE_ROOT = activity.get_bundle_path()
    SUGAR = True
except ImportError:
    MEASURE_ROOT = os.environ['HOME']
    SUGAR = False

from gettext import gettext as _

ICONS_DIR = os.path.join(MEASURE_ROOT, 'icons')

# Multiplied with width and height to set placement of text
TEXT_X_M = 0.65
TEXT_Y_M = 0.70

# Maximum number of graphs that can be simultaneously be displayed
MAX_GRAPHS = 4

# Device settings at start of Activity
RATE = 48000
MIC_BOOST = True
DC_MODE_ENABLE = False
CAPTURE_GAIN = 50
BIAS = True

# Interval, in ms, after which audio buffer will be sent to drawing class
AUDIO_BUFFER_TIMEOUT = 30

# When Activity quits
QUIT_MIC_BOOST = False
QUIT_DC_MODE_ENABLE = False
QUIT_CAPTURE_GAIN = 100
QUIT_BIAS = True

# Maximum no. of data samples Measure will save
MAX_LOG_ENTRIES = 1000

# Duty cycle of display value update
DISPLAY_DUTY_CYCLE = 100

# Hardware configurations
XO1 = 'XO-1'
XO15 = 'XO-1.5'
XO175 = 'XO-1.75'
XO4 = 'XO-4'
NL3 = 'NL3'
UNKNOWN = 'unknown'

# Bounds of side slider
LOWER = 0.0
UPPER = 4.0

# TRANS: Please use short versions of instrument names if possible
INSTRUMENT_DICT = {
    _('None'): [],
    _('Guitar'): [82.4069, 110, 146.832, 195.998, 246.942, 329.628],
    _('Violin'): [195.998, 293.665, 440, 659.255],
    _('Viola'): [130.813, 195.998, 293.665, 440],
    _('Cello'): [65.4064, 97.9989, 146.832, 220],
    _('Bass'): [41.2034, 55, 73.4162, 97.9989],
    # TRANS: a small Andean stringed instrument of the lute family
    _('Charango'): [329.63, 392, 440, 523.25, 659.26],
    # TRANS: a small string instrument (also called machimbo)
    _('Cavaquinho'): [293.665, 391.995, 493.883, 587.330],
    _('Ukulele'): [261.626, 329.628, 391.995, 440],
    _('Sitar'): [174.614, 130.813, 195.998, 65.4064, 391.995,
                 261.626, 523.251],
    _('Mandolin'): [195.998, 293.665, 440, 659.255],
    _('Recorder'): [391.995]}
